src/tests/utests/xi_utest_data_desc.c

```diff
@@ -244,6 +244,259 @@ XI_TT_TESTCASE_WITH_SETUP(
         tt_fail();
     } )
 
+#ifdef XI_MEMORY_LIMITER_ENABLED
+XI_TT_TESTCASE(
+    utest__xi_make_desc_from_string_copy__valid_data__not_enough_memory_for_buffer,
+    {
+        /* size values */
+        size_t num_characters     = 700;
+        xi_state_t ret_state      = XI_STATE_OK;
+        char* origin_string       = NULL;
+        xi_data_desc_t* data_desc = NULL;
+        
+        const size_t allocation_space   = num_characters * sizeof( char ) +
+                                          sizeof( xi_data_desc_t ) +
+                                          sizeof( xi_data_desc_t ) +
+                                          2*sizeof( xi_memory_limiter_entry_t );
```

Comment:
for first and second sight this looks magic. What are the components `allocation_space` is built up?

Reply:
OK, sure!  I can create a variable for each one individually.